storage/rawstores.go

```diff
@@ -84,3 +84,28 @@ func NewCompositeDataStore(refConstructor ReferenceConstructor, composedProtobuf
 		ComposedProtobufStore: composedProtobufStore,
 	}
 }
+
+func (ds *DataStore) RefreshConfig(cfg *Config) error {
```

Comment:
sorry one last request, @pmahindrakar-oss mentioned that this isn't thread-safe which is a great catch - do you mind adding a comment saying so?